tests/src/roomHandlers/fishingRoomHandler.test.ts

```diff
@@ -0,0 +1,157 @@
+
+const roomType: RoomType = 'fishingRoom';
+type TRoomType = typeof roomType;
+type TArrangedTransitionData = ArrangedTransitionData<TRoomType>;
+
+testSingletonModule(
+  'roomHandlers/FishingRoomHandler',
+  ({ testIntegration }) => {
+    testIntegration('run', ({ testScenario }) => {
+      const {
+        testDefaultCommandAtEntrance,
+        testInvalidCommandAtEntrance,
+        testStateTransition,
+      } = buildStateTransitionHelpers(
+        testScenario,
+        roomType,
+        new FishingRoomHandler(),
+      );
+
+      testDefaultCommandAtEntrance();
+
+      testInvalidCommandAtEntrance();
+
+      testStateTransition(
+        'AtEntrance --> AtEntrance: fix',
+        (): TArrangedTransitionData => ({
+          startingRoomState: {
+            fishCaught: 0,
+            isRodBroken: true,
+            randomNumber: 0,
+          },
+          stateAssertionDescription: 'updates the room state',
+          expectedResult: {
+            commandDescription: 'You fix your fishing line.',
+            roomState: {
+              fishCaught: 0,
+              isRodBroken: false,
+              randomNumber: 0,
+            },
+          },
+        }),
+      );
+
+      testStateTransition(
+        'AtEntrance --> AtEntrance: fish',
+        (): TArrangedTransitionData => ({
+          startingRoomState: {
+            fishCaught: 0,
+            isRodBroken: true,
+            randomNumber: 0,
+          },
+          stateAssertionDescription: 'does not update the room state',
+          expectedResult: {
+            commandDescription:
+              'You sit down and start to get ready to fish. But your rod is broken and you have to fix it.',
+            roomState: {
+              fishCaught: 0,
+              isRodBroken: true,
+              randomNumber: 0,
+            },
+          },
+        }),
+      );
+
+      testStateTransition(
+        'AtEntrance --> Fishing: fish',
+        (): TArrangedTransitionData => ({
+          startingRoomState: {
+            fishCaught: 0,
+            isRodBroken: false,
+            randomNumber: 0,
+          },
+          stateAssertionDescription: null,
+          expectedResult: {
+            commandDescription:
+              'You sit down and start to get ready to fish. This is going to be alot of fun!',
+            roomState: {
+              fishCaught: 0,
+              isRodBroken: false,
+              randomNumber: 0,
+            },
+          },
+        }),
+      );
+
+      testStateTransition(
+        'Fishing --> Fishing: continue',
+        (): TArrangedTransitionData => ({
+          startingRoomState: {
+            fishCaught: 0,
+            isRodBroken: false,
+            randomNumber: 0,
+          },
+          stateAssertionDescription: 'does not update the room state',
+          expectedResult: {
+            commandDescription:
+              'You continue fishing...\n        ... but nothing happens\n        \n        ',
```

Comment:
The newlines are on purpose. I'm fine with leaving the tests like this, but I can update `testStateTransition` to trim the newlines if you want.

Reply:
If the new lines are intentional then I don't think the test helper should modify anything. My only question is did you set the whitespace so the test output would line up?

This is what it looks like in the test:

![image](https://user-images.githubusercontent.com/3934393/159772082-3c2e0dd8-b762-4a93-8f92-7d04db8d87dd.png)


And in the console:

![image](https://user-images.githubusercontent.com/3934393/159772182-1f925254-6e83-4552-a0ce-331589b51e1d.png)

Reply:
After taking a look at the whitespace I'm really confused by the expected behavior. Each result has varying before/after whitespace.